autoPyTorch/pipeline/components/training/data_loader/time_series_forecasting_data_loader.py

```diff
@@ -0,0 +1,592 @@
+        # for instance, if we have a train indices is expressed as [0, 1, 2 ,3, 7 ,8 ].
+        #  A new sequence must start from the index 7. We could then split each unique values to represent the length
+        # of each split
+
+        # TODO consider min_start as a hp (multiple of self.n_prediction_steps?)
+        min_start = self.n_prediction_steps
+
+        dataset_seq_length_train_all = X['dataset_properties']['sequence_lengths_train']
+        if np.sum(dataset_seq_length_train_all) == len(train_split):
+            # this applies if we want to fit the entire datasets
+            seq_train_length = np.array(dataset_seq_length_train_all)
+        else:
+            _, seq_train_length = np.unique(train_split - np.arange(len(train_split)), return_counts=True)
+
+        num_instances_per_seqs = self.compute_expected_num_instances_per_seq(num_instances_dataset,
+                                                                             seq_train_length,
+                                                                             min_start,
+                                                                             fraction_seq,
+                                                                             fraction_samples_per_seq,
+                                                                             )
+
+        # TODO consider the case where num_instances_train is greater than num_instances_dataset,
+        # In which case we simply iterate through all the datasets
+
+        sampler_indices_train = np.arange(num_instances_dataset)
+
+        self.sampler_train = TimeSeriesSampler(indices=sampler_indices_train, seq_lengths=seq_train_length,
+                                               num_instances_per_seqs=num_instances_per_seqs,
+                                               min_start=min_start)
+
+        self.train_data_loader = torch.utils.data.DataLoader(
+            train_dataset,
+            batch_size=min(self.batch_size, len(sampler_indices_train)),
+            shuffle=False,
+            num_workers=X.get('num_workers', 0),
+            pin_memory=X.get('pin_memory', True),
+            drop_last=X.get('drop_last', True),
+            collate_fn=partial(custom_collate_fn, x_collector=self.padding_collector),
+            sampler=self.sampler_train,
+        )
+
+        # validation set is not so important here, we make  the size of validation set to be 20% of training instances
+        num_samples_val = int(np.sum(num_instances_per_seqs)) // 5
+        if num_samples_val > len(val_dataset):
+            sampler_val = None
+        else:
+            sampler_val = SequentialSubSetSampler(data_source=val_dataset,
+                                                  num_samples=num_samples_val)
+
+        self.val_data_loader = torch.utils.data.DataLoader(
+            val_dataset,
+            batch_size=min(1000, len(val_dataset)),
```

Comment:
yes, but it is possible for larger datasets, a large batch size means we are not able to fit in the memory. And using the hyperparameter allows us to be sure that it does, else the configuration will have already failed. Moreover, it will also allow the user to pass the batch size according to their needs, so they can set a larger batch size if they need. With your current change, it is pretty much hard coded. What do you think?